packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/sh
+
+# set SYSCONFDIR to /etc if it is not set
+if [ -z "$SYSCONFDIR" ]; then
+    SYSCONFDIR=/etc
+fi
+
+# see if /etc/rhsm/rhsm.conf exists
+RHSMCONF_PATH=$SYSCONFDIR/rhsm/rhsm.conf
+if [ -f $RHSMCONF_PATH ]; then
+    echo "Getting configuration from $RHSMCONF_PATH..."
+    # get cert-file and key-file from rhsm.conf
+    CONSUMER_CERT_DIR=$(cat ${RHSMCONF_PATH} | grep consumerCertDir | cut -f3 -d ' ')
```

Comment:
```suggestion
    CONSUMER_CERT_DIR=$(grep consumerCertDir ${RHSMCONF_PATH} | cut -f3 -d ' ')
```

no need to `cat` ;)

`rhsm.conf` is an ini-style file, which can have comments, and someone could have added a `# consumerCertDir is for consumer certs` before the actual assignment line, brealing your `grep|cut` logic.

The following *should* be safer: `grep '^consumerCertDir' /etc/rhsm/rhsm.conf | cut -d= -f2 | xargs` (only look for lines *starting* with `consumerCertDir`, split at `=`, and the `xargs` removes any extra whitespace.

This still would fail if there is an inline comment though…

Reply:
I love overengineering! :smile:

Reply:
Why stop now?

If we're concerned about the file having been touched, subscription-manager can output a cleaned up version: `subscription-manager config --list`

Reply:
Are you saying you want to overwrite `rhsm.conf` with this output? But then we'd lose any comments, a lot of which are useful explanations of the settings.

Reply:
I'm only saying it's possible to read a version of rhsm.conf in a seemingly standard format unaffected by comments, spacing, etc that the user has done

Reply:
oohh, gotcha!  :thinking:

Reply:
I think I'm going to go this route rather than rewriting the entire thing in Python.  Seems easier.